pcp/Platform.c

```diff
@@ -0,0 +1,874 @@
+/*
+htop - linux/Platform.c
+(C) 2014 Hisham H. Muhammad
+(C) 2020-2021 htop dev team
+(C) 2020-2021 Red Hat, Inc.  All Rights Reserved.
+Released under the GNU GPLv2, see the COPYING file
+in the source distribution for its full text.
+*/
+
+#include "config.h"
+
+#include <math.h>
+
+#include "BatteryMeter.h"
+#include "ClockMeter.h"
+#include "Compat.h"
+#include "CPUMeter.h"
+#include "DateMeter.h"
+#include "DateTimeMeter.h"
+#include "DiskIOMeter.h"
+#include "HostnameMeter.h"
+#include "LoadAverageMeter.h"
+#include "Macros.h"
+#include "MainPanel.h"
+#include "MemoryMeter.h"
+#include "Meter.h"
+#include "NetworkIOMeter.h"
+#include "Object.h"
+#include "Panel.h"
+#include "PCPProcess.h"
+#include "PCPProcessList.h"
+#include "Platform.h"
+#include "ProcessList.h"
+#include "ProvideCurses.h"
+#include "Settings.h"
+#include "SwapMeter.h"
+#include "SysArchMeter.h"
+#include "TasksMeter.h"
+#include "UptimeMeter.h"
+#include "XUtils.h"
+#include "linux/PressureStallMeter.h"
+#include "linux/ZramMeter.h"
+#include "linux/ZramStats.h"
+#include "zfs/ZfsArcMeter.h"
+#include "zfs/ZfsArcStats.h"
+#include "zfs/ZfsCompressedArcMeter.h"
+
+
+typedef struct Platform_ {
+   int context;			/* PMAPI(3) context identifier */
+   unsigned int total;		/* total number of all metrics */
+   const char** names;		/* name array indexed by Metric */
+   pmID* pmids;			/* all known metric identifiers */
+   pmID* fetch;			/* enabled identifiers for sampling */
+   pmDesc* descs;		/* metric desc array indexed by Metric */
+   pmResult* result;		/* sample values result indexed by Metric */
```

Comment:
IWYU violation: which header provides these types? Should be included as part of this file (unless part of public interface, in which case the public interface header should include these).

Reply:
Its <pcp/pmapi.h> - which arrives via pcp/Platform.h ... seems strange to have to include it in the pcp/Platform.c as well...?

Reply:
If it's included via `pcp/Platform.h` it's okay, to drop it in the `.c`.